### PR TITLE
record minted rewards between lockups

### PR DESCRIPTION
**Pull request by mkurnikov**
@movekevin

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2308)

### CODE REVIEW

**Review by davidiw — COMMENTED** [review]
@movekevin

**Comment by mkurnikov** [comment]
@movekevin Hi, do you think it's mergeable? I'll get it to completion then.

**Comment by mkurnikov** [comment]
@movekevin Hi, do you think it's mergeable? I'll get it to completion then.

File: aptos-move/framework/aptos-framework/sources/configs/stake.move [hunk 1]
```diff
@@ -674,6 +684,10 @@ module aptos_framework::stake {
         let old_locked_until_secs = stake_pool.locked_until_secs;
         stake_pool.locked_until_secs = new_locked_until_secs;
 
+        if (old_locked_until_secs <= timestamp::now_seconds()) {
+            stake_pool.rewards_minted = 0;
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/configs/stake.move:688]: This doesn't seem right. Why is this only updated here? The lockup could have expired and the validator might not call increase_lockup or increase_lockup_with_cap

**Comment by davidiw** [comment]
closing out, feel free to reopen if you'd like to wrap it up.

**Pull request closed without merging**